File: plugin/src/main/java/io/j99/idea/vue/VueFileTypeFactory.java
package io.j99.idea.vue;

import com.intellij.openapi.fileTypes.FileTypeConsumer;
import com.intellij.openapi.fileTypes.FileTypeFactory;
import org.jetbrains.annotations.NotNull;

/**
 * Created by apple on 16/2/26.
 */
public class VueFileTypeFactory extends FileTypeFactory {
    @Override
    public void createFileTypes(@NotNull FileTypeConsumer fileTypeConsumer) {
        for (String type:VueFileType.DEFAULT_ASSOCIATED_EXTENSIONS) {
            fileTypeConsumer.consume(VueFileType.INSTANCE, type);
        }
    }
}

File: src/main/kotlin/io/j99/idea/vue/VueFileTypeFactory.kt
package io.j99.idea.vue

import com.intellij.openapi.fileTypes.FileTypeConsumer
import com.intellij.openapi.fileTypes.FileTypeFactory

/**
 * Created by android on 15-12-25.
 */
class VueFileTypeFactory : FileTypeFactory() {
    override fun createFileTypes(fileTypeConsumer: FileTypeConsumer) {
        for (i in VueFileType.DEFAULT_ASSOCIATED_EXTENSIONS.indices) {
            fileTypeConsumer.consume(VueFileType.INSTANCE, VueFileType.DEFAULT_ASSOCIATED_EXTENSIONS[i])
        }
    }
}
